### ISSUE TITLE
Prebuilt APKs

### ISSUE THREAD

**Issue by Mennaruuk** [galaxygoldfish/pineapple#1]
Are there prebuilt APKs for us noobs?

**Comment by galaxygoldfish** [comment]
I was planning on releasing it to the Play Store but unfortunately that won't be happening now that the Reddit API is a paid service, and if I built an APK I would have to release my personal API key. You can make a developer account on Reddit and get your own key and then build the APK for your personal use, however.

**Comment by TeaOwO** [comment]
> I was planning on releasing it to the Play Store but unfortunately that won't be happening now that the Reddit API is a paid service, and if I built an APK I would have to release my personal API key. You can make a developer account on Reddit and get your own key and then build the APK for your personal use, however. 

Problem is I personally get an error when I build the app.

**Comment by Mennaruuk** [comment]
> > I was planning on releasing it to the Play Store but unfortunately that won't be happening now that the Reddit API is a paid service, and if I built an APK I would have to release my personal API key. You can make a developer account on Reddit and get your own key and then build the APK for your personal use, however.
> 
> Problem is I personally get an error when I build the app. (Using Android studio)

Me too. Any chance it's related to the Kotlin version?

**Comment by TeaOwO** [comment]
> > > I was planning on releasing it to the Play Store but unfortunately that won't be happening now that the Reddit API is a paid service, and if I built an APK I would have to release my personal API key. You can make a developer account on Reddit and get your own key and then build the APK for your personal use, however.
> > 
> > Problem is I personally get an error when I build the app. (Using Android studio)
> 
> Me too. Any chance it's related to the Kotlin version?

I don't know, haven't been able to solve it.

**Comment by TeaOwO** [comment]
> > > I was planning on releasing it to the Play Store but unfortunately that won't be happening now that the Reddit API is a paid service, and if I built an APK I would have to release my personal API key. You can make a developer account on Reddit and get your own key and then build the APK for your personal use, however.
> > 
> > Problem is I personally get an error when I build the app. (Using Android studio)
> 
> Me too. Any chance it's related to the Kotlin version?

I seem to have fixed it by using the Gradle command line.

 ```
.\gradlew.bat assembleDebug
```

**Comment by Donkey-Doug** [comment]
What about haping people enter their own API key after launching the app?

**Comment by devlocalhost** [comment]
> What about haping people enter their own API key after launching the app?

yes please, an option in settings because building this is TOO HARD, theres no clear steps on how to setup all this, installing the build tools, sdk manager and whatever shit is needed, ive been trying for more than 3 hours, ive surfed the whole internet but im STILL struggling and i DONT want to use the bloat android studio app cause i dont have a supercomputer to run all that

**Comment by galaxygoldfish** [comment]
> > What about haping people enter their own API key after launching the app?
> 
> yes please, an option in settings because building this is TOO HARD, theres no clear steps on how to setup all this, installing the build tools, sdk manager and whatever shit is needed, ive been trying for more than 3 hours, ive surfed the whole internet but im STILL struggling and i DONT want to use the bloat android studio app cause i dont have a supercomputer to run all that

I'm really busy right now but if someone would like to implement this feature and make a pull request, I'll happily merge it! As for building the APK, the easiest way to do so is to install Android studio, import the project and click the Run button. This ensures that you have all of the dev tools and plugins necessary to build since they'll be installed during the setup process

**Comment by devlocalhost** [comment]
> the easiest way to do so is to install Android studio, import the project and click the Run button. This ensures that you have all of the dev tools and plugins necessary to build since they'll be installed during the setup process

It's not only easy, but also painful and "expensive". (Hardware side). Isn't android studio heavy?

**Comment by creepy-pasta101** [comment]
> I'm really busy right now but if someone would like to implement this feature and make a pull request, I'll happily merge it! As for building the APK, the easiest way to do so is to install Android studio, import the project and click the Run button. This ensures that you have all of the dev tools and plugins necessary to build since they'll be installed during the setup process

@galaxygoldfish I was able to implement the feature [here](https://github.com/suhan-paradkar/pineapple/tree/clientsecret)......
Will be opening a PR once #6 is merged

P.S. That branch is dependent on branch of #6

**Comment by galaxygoldfish** [comment]
> > the easiest way to do so is to install Android studio, import the project and click the Run button. This ensures that you have all of the dev tools and plugins necessary to build since they'll be installed during the setup process
> 
> It's not only easy, but also painful and "expensive". (Hardware side). Isn't android studio heavy?

It actually only requires about 8GB of disk space to install and then the recommended RAM amount is 8GB 

Details about minimum requirements -https://developer.android.com/studio/install
Link to download AS - https://developer.android.com/studio

**Comment by galaxygoldfish** [comment]
> > > I was planning on releasing it to the Play Store but unfortunately that won't be happening now that the Reddit API is a paid service, and if I built an APK I would have to release my personal API key. You can make a developer account on Reddit and get your own key and then build the APK for your personal use, however.
> > 
> > 
> > Problem is I personally get an error when I build the app. (Using Android studio)
> 
> Me too. Any chance it's related to the Kotlin version?

I'm using AS Hedgehog 2023.1.1 Canary 5, and just merged a PR that updates dependencies, and it looks like for me at least, the build errors have been resolved. Make sure you also have all of the SDK updates installed too 👍

**Comment by devlocalhost** [comment]
> > > the easiest way to do so is to install Android studio, import the project and click the Run button. This ensures that you have all of the dev tools and plugins necessary to build since they'll be installed during the setup process
> > 
> > 
> > It's not only easy, but also painful and "expensive". (Hardware side). Isn't android studio heavy?
> 
> It actually only requires about 8GB of disk space to install and then the recommended RAM amount is 8GB
> 
> Details about minimum requirements -https://developer.android.com/studio/install Link to download AS - https://developer.android.com/studio

I have the disk space but I don't have enough RAM. It is possible to build it without using android studio, but for some reason it can't find the sdk, even though I have everything set up correctly (I think)

**Comment by idontveapc** [comment]
did it happen? did anyone build apk(in which we can put own api)?

**Comment by creepy-pasta101** [comment]
> did it happen? did anyone build apk(in which we can put own api)?

You will be able to that once #7 is merged

**Comment by BabyOilJohnson** [comment]
#7 got merged. Now what to do?

**Comment by devlocalhost** [comment]
> [#7](https://github.com/galaxygoldfish/pineapple/pull/7) got merged. Now what to do?

You build the app yourself. Check the [readme](https://github.com/galaxygoldfish/pineapple?tab=readme-ov-file#build-notes)

**Comment by m-obeid** [comment]
whats the redirect uri for the key, i tried pineapple://login but it just throws net::ERR_UNKNOWN_URL_SCHEME

**Comment by m-obeid** [comment]
nevermind, it just doesnt work in webview

**Comment by BabyOilJohnson** [comment]
@galaxygoldfish, can we expect an APK release, please?

**Comment by Humber555** [comment]
Can you use the RedReader API key to give us a pre-built APK please?

**Comment by galaxygoldfish** [comment]
Let me know if this works. API key can be entered by user during setup 

[app-debug.apk.zip](https://github.com/user-attachments/files/30197837/app-debug.apk.zip)

**Comment by Humber555** [comment]
I tried using the RedReader api key, but I think the redirect URI has to be redreader://rr_oauth_redir , beacause it won't work if I use it. I don't know how does this work, but other clients like Ilay for Reddit are working with this method. When you are proceeding to the setup, you have to enter the API key and the redirect URI, and then I enter the one from RedReader and it works.

**Comment by BabyOilJohnson** [comment]
Yes, please provide support for using the RedReader API key, and thank you, dev, for the APK!
